### PR TITLE
Add "Enable All" to context menu of Breakpoints view

### PR DESCRIPTION
**Pull request by trancexpress**
Added Breakpoints view "Enable All" breakpoints action, similar to the existing "Disable All" breakpoints action.

Fixes: #198

### CODE REVIEW

File: org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/AllBreakpointsEnablementAction.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		}
+		IWorkbenchWindow window = DebugUIPlugin.getActiveWorkbenchWindow();
+		if (window == null) {
+			return;
+		}
+		IPreferenceStore store = DebugUIPlugin.getDefault().getPreferenceStore();
+		boolean prompt = store.getBoolean(this.fPreferenceName);
+		boolean proceed = true;
+		if (prompt) {
+			MessageDialogWithToggle mdwt = MessageDialogWithToggle.openYesNoQuestion(window.getShell(),
```
Comment (trancexpress) [inline org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/AllBreakpointsEnablementAction.java:99]: I think I'll "merge the preference and dialog into one, enabling and disabling all breakpoints.

File: org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/AllBreakpointsEnablementAction.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+					ActionMessages.AllBreakPointsActionEnablement_DontAskAgain, !prompt, null, null);
+			if (mdwt.getReturnCode() != IDialogConstants.YES_ID) {
+				proceed = false;
+			} else {
+				store.setValue(IDebugPreferenceConstants.PREF_PROMPT_DISABLE_ALL_BREAKPOINTS,
+						!mdwt.getToggleState());
+			}
+		}
+		if (proceed) {
+			new Job(this.fLabel2) {
```
Comment (lbownik) [inline org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/AllBreakpointsEnablementAction.java:109]: how about wrapping the body of this if in a private method (like "schesuleJob") and then call it in line 102?
"proceed" variable could  be removed afterwards

File: org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/AllBreakpointsEnablementAction.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+				protected IStatus run(IProgressMonitor monitor) {
+					try {
+						for (IBreakpoint breakpoint : breakpoints) {
+							breakpoint.setEnabled(fEnable);
+						}
+					} catch (CoreException e) {
+						DebugUIPlugin.log(e);
+						return Status.CANCEL_STATUS;
+					}
+					return Status.OK_STATUS;
```
Comment (lbownik) [inline org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/AllBreakpointsEnablementAction.java:120]: wouldn't it be more readable if "return Status.OK_STATUS;" was moved to the end of try block (between lines 115 and 116)?

**Comment by laeubi** [comment]
This repository was merged with https://github.com/eclipse-platform/eclipse.platform if this PR is still relevant please rebase it and push it to the new origin!

**Pull request closed without merging**